test/mbedclient/utest/nsdlaccesshelper/test_nsdlaccesshelper.cpp

```diff
@@ -105,34 +105,34 @@ Test_NsdlAccessHelper::~Test_NsdlAccessHelper()
 {
     delete observer;
     observer = NULL;
+
+    clear_list();
 }
 
 void Test_NsdlAccessHelper::test_nsdl_c_callback()
 {
-
     CHECK(__nsdl_c_callback(NULL,NULL,NULL,SN_NSDL_PROTOCOL_HTTP) == 0 );
 
     m2mnsdlinterface_stub::int_value = 1;
-    __nsdl_interface = new M2MNsdlInterface(*observer);
+    m2mnsdlinterface_stub::void_value = malloc(1);
+    __nsdl_interface_list.push_back(new M2MNsdlInterface(*observer));
 
-    CHECK(__nsdl_c_callback(NULL,NULL,NULL,SN_NSDL_PROTOCOL_HTTP) == 1 );
-
-    delete __nsdl_interface;
-    __nsdl_interface = NULL;
+    CHECK(__nsdl_c_callback((nsdl_s*)m2mnsdlinterface_stub::void_value,
+                            NULL,NULL,SN_NSDL_PROTOCOL_HTTP) == 1 );
+    free(m2mnsdlinterface_stub::void_value);
+    clear_list();
 }
 
 void Test_NsdlAccessHelper::test_nsdl_c_memory_alloc()
 {
     void *ptr = __nsdl_c_memory_alloc(6);
-    CHECK(ptr == NULL);
-
-    __nsdl_interface = new M2MNsdlInterface(*observer);
-    ptr = __nsdl_c_memory_alloc(6);
     CHECK(ptr != NULL);
-
-    delete __nsdl_interface;
-    __nsdl_interface = NULL;
     free(ptr);
+    ptr = NULL;
+    ptr = __nsdl_c_memory_alloc(UINT16_MAX+1);
+    CHECK(ptr == NULL);
+
+//free(ptr);
```

Comment:
Comment removed. However freeing in this case is not needed since the allocation fails in above step.